pytensor/tensor/variable.py

```diff
@@ -438,6 +438,62 @@ def trunc(self):
     def astype(self, dtype):
         return pt.basic.cast(self, dtype)
 
+    def _getitem_with_newaxis(self, args):
+        """Handle newaxis (None) for both basic and advanced indexing.
+
+        `np.newaxis` (i.e. `None`) in NumPy indexing mean "add a new
+        broadcastable dimension at this location".  Since PyTensor adds
+        new broadcastable dimensions via the `DimShuffle` `Op`, the
+        following code uses said `Op` to add one of the new axes and
+        then uses recursion to apply any other indices and add any
+        remaining new axes.
+        """
+        counter = 0
+        pattern = []
+        new_args = []
+        for arg in args:
+            if arg is np.newaxis or arg is NoneConst:
+                pattern.append("x")
+                new_args.append(slice(None))
+            else:
+                # Check for boolean index which consumes multiple dimensions
```

Comment:
This is probably right, but why does it matter that boolean indexing consumes multiple dimensions? Aren't we doing `expand_dims where there was None -> replace new_axis by None slice -> index again`?